### PR TITLE
feat: add base components 🧱

### PR DESCRIPTION
**Pull request by doranakan**
### Description
This PR adds the foundational components to kickstart the project's design system, including the `Button` and the `Card` component. Additionally, I have introduced a reusable `InnerBorder` component to enhance the glass effect on the `Card`.

### Changes:
1. **Button Component:**
   - Built on top of `Pressable`, with two variants:
     - One variant for handling Expo Router links.
     - Another for normal callbacks.
   
2. **Card Component:**
   - A styled container featuring:
     - A gradient background.
     - A shadow effect.
     - Glass effect, achieved using the reusable `InnerBorder` component.

### Cross-platform Compatibility:
- The components have been tested and work as expected on both mobile and web platforms.

### CODE REVIEW

File: components/Button/Button.tsx [hunk 1]
```diff
@@ -1,24 +1,72 @@
...
-        as={Skull}
+        as={icon}
         size='xl'
       />
-    </Pressable>
-  )
-}
-
+    ) : null}
+    {text ? <Text className='text-typography-50'>{text}</Text> : null}
```
Comment (lucacaputo) [inline components/Button/Button.tsx:69]: both text and icon are possibly undefined, could we end up with an empty HStack?

File: components/Card/Card.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import type { Meta, StoryObj } from '@storybook/react'
+import { StoryContainer } from '../storybook'
```
Comment (lucacaputo) [inline components/Card/Card.stories.tsx:2]: My IDE (🤡 ) complains about casing in this import and all other occurrences, we should use the exact name of the directory (`../StoryBook`)

File: components/Button/Button.tsx (hunk 1 above)
Comment (doranakan) [reply components/Button/Button.tsx:69]: correcg, we should fix the type of this component by making at least one of these fields requires

File: components/Card/Card.stories.tsx (hunk 2 above)
Comment (doranakan) [reply components/Card/Card.stories.tsx:2]: that's just a typo, let's rename it using lowercase

File: components/Card/Card.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+const Card = ({ children }: CardProps) => (
+  <VStack className='w-full rounded-2xl bg-primary-700 shadow-lg'>
+    <InnerBorder>
+      {/* <LinearGradient
+        className='bg-gradient-to-br from-primary-700 to-primary-900'
+        colors={['bg-primary-700', 'bg-primary-900']}
+        start={{ x: 0.3, y: 1 }}
+        end={{ x: 1, y: 3 }}
+      >
+      </LinearGradient> */}
```
Comment (lucacaputo) [inline components/Card/Card.tsx:16]: shall we uncomment/remove this?

**Comment by Viraj-10** [comment]
Hi @doranakan,  

All of our components follow a well-implemented design system with standard best practices. You can explore the design files and guidelines [here](https://www.figma.com/community/file/1358053104938234615).  

If you have any questions or need assistance with the design, feel free to start a discussion—we’re here to help!  

Regarding your suggestions, we understand they are based on your specific use case. That’s why we provide copy-pastable patterns that allow users to easily customize them to suit their needs.  


The discussion around using `expo-linear-gradient` might feel a bit biased since not everyone may choose to use it. Some users might already have `react-native-skia` installed and prefer to use that instead. You can check out more about `react-native-skia` gradients [here](https://shopify.github.io/react-native-skia/docs/shaders/gradients/#linear-gradient).  

While we've provided a recipe for `expo-linear-gradient`, we understand that we can't (and shouldn't) force users to follow a specific approach. We want to give developers the flexibility to choose what works best for their project.

**Comment by doranakan** [comment]
> All of our components follow a well-implemented design system with standard best practices. You can explore the design files and guidelines [here](https://www.figma.com/community/file/1358053104938234615).
> 
> If you have any questions or need assistance with the design, feel free to start a discussion—we’re here to help!
> 
> Regarding your suggestions, we understand they are based on your specific use case. That’s why we provide copy-pastable patterns that allow users to easily customize them to suit their needs.
> 
> The discussion around using `expo-linear-gradient` might feel a bit biased since not everyone may choose to use it. Some users might already have `react-native-skia` installed and prefer to use that instead. You can check out more about `react-native-skia` gradients [here](https://shopify.github.io/react-native-skia/docs/shaders/gradients/#linear-gradient).
> 
> While we've provided a recipe for `expo-linear-gradient`, we understand that we can't (and shouldn't) force users to follow a specific approach. We want to give developers the flexibility to choose what works best for their project.

Thank you for the detailed response and the flexibility you’re aiming to provide with customizable patterns!

I agree with your point about allowing users to choose the approach that works best for their project, whether it’s expo-linear-gradient or react-native-skia. However, it seems like the issue with NativeWind colors would persist regardless of the method suggested, even with the custom patterns.

Do you have any recommendations on how we could manually select or override NativeWind colors at runtime? This would help developers like myself better adapt the components to specific needs, especially when custom runtime color management is required.

Looking forward to hearing your thoughts!

**Comment by Viraj-10** [comment]
you can directly use token values. 

Changing your LinearGradient to this will work on native device.

```

'use client';
import React from 'react';
import { tva } from '@gluestack-ui/nativewind-utils/tva';
import { LinearGradient as ExpoLinearGradient } from 'expo-linear-gradient';
import { cssInterop } from 'nativewind';

cssInterop(ExpoLinearGradient, {
  className: 'style',
});

const linearGradientStyle = tva({
  base: '',
});

export const LinearGradient = React.forwardRef(
  ({ className, from, to, ...props }: any, ref?: any) => {

    return (
      <ExpoLinearGradient
        {...props}
        colors={[from, to]}
        ref={ref}
      />
    );
  }
);

cssInterop(LinearGradient, {
  from:{
    target:"from",
    nativeStyleToProp:{
      color:true,
    }
  },
  to:{
    target:"to",
    nativeStyleToProp:{
      color:true,
    }
  }
});
```
```
 <LinearGradient
           className="w-full rounded-full items-center py-2"
            from="color-success-500"
            to="color-primary-600"
            start={[0, 1]}
            end={[1, 0]}
          />
```
for web you can have custom implementation with `index.web.tsx` with `<div />` 

For more info on cssInterop you can have look at `https://www.nativewind.dev/v4/api/css-interop`

**Comment by doranakan** [comment]
> you can directly use token values.
> 
> Changing your LinearGradient to this will work on native device.
> 
> ```
> 
> 'use client';
> import React from 'react';
> import { tva } from '@gluestack-ui/nativewind-utils/tva';
> import { LinearGradient as ExpoLinearGradient } from 'expo-linear-gradient';
> import { cssInterop } from 'nativewind';
> 
> cssInterop(ExpoLinearGradient, {
>   className: 'style',
> });
> 
> const linearGradientStyle = tva({
>   base: '',
> });
> 
> export const LinearGradient = React.forwardRef(
>   ({ className, from, to, ...props }: any, ref?: any) => {
> 
>     return (
>       <ExpoLinearGradient
>         {...props}
>         colors={[from, to]}
>         ref={ref}
>       />
>     );
>   }
> );
> 
> cssInterop(LinearGradient, {
>   from:{
>     target:"from",
>     nativeStyleToProp:{
>       color:true,
>     }
>   },
>   to:{
>     target:"to",
>     nativeStyleToProp:{
>       color:true,
>     }
>   }
> });
> ```
> 
> ```
>  <LinearGradient
>            className="w-full rounded-full items-center py-2"
>             from="color-success-500"
>             to="color-primary-600"
>             start={[0, 1]}
>             end={[1, 0]}
>           />
> ```
> 
> for web you can have custom implementation with `index.web.tsx` with `<div />`
> 
> For more info on cssInterop you can have look at `https://www.nativewind.dev/v4/api/css-interop`

thanks @Viraj-10 for the suggestion. this way we are capable of display the linear gradient correctly with tailwind colors

**Pull request merged**